cdap-ui/app/cdap/components/PipelineConfigurations/ConfigurationsContent/PipelineConfigTabContent/index.js

```diff
@@ -0,0 +1,264 @@
+        </UncontrolledTooltip>
+      </div>
+    </div>
+  );
+};
+
+StageLogging.propTypes = {
+  stageLogging: PropTypes.bool,
+  onToggle: PropTypes.func
+};
+
+const ConnectedStageLogging = connect(
+  mapStateToStageLoggingProps,
+  mapDispatchToStageLoggingProps
+)(StageLogging);
+
+const mapStateToCheckpointingProps = (state) => {
+  return {
+    checkpointing: state.disableCheckpoints
+  };
+};
+const mapDispatchToCheckpointingProps = (dispatch) => {
+  return {
+    onToggle: (value) => {
+      dispatch({
+        type: PipelineConfigurationsActions.SET_CHECKPOINTING,
+        payload: { checkpointing: value }
+      });
+    }
+  };
+};
+
+const Checkpointing = ({checkpointing, onToggle}) => {
+  return (
+    <div className="label-with-toggle checkpointing row">
+      <span className="toggle-label col-xs-4">Checkpointing</span>
+      <div className="col-xs-7 toggle-container">
+        <ToggleSwitch
+          isOn={checkpointing}
+          onToggle={onToggle.bind(null, !checkpointing)}
+        />
+        <IconSVG
+          name="icon-info-circle"
+          id="checkpointing-info-icon"
+        />
+        <UncontrolledTooltip
+          target="checkpointing-info-icon"
+          delay={{show: 250, hide: 0}}
+          placement="right"
+        >
+          Allows Apache Spark Streaming to checkpoint data (RDDs) to persistent storage so that the pipeline can recover from failures.
+        </UncontrolledTooltip>
+      </div>
+    </div>
+  );
+};
+
+Checkpointing.propTypes = {
+  checkpointing: PropTypes.bool,
+  onToggle: PropTypes.func
+};
+
+const ConnectedCheckpointing = connect(
+  mapStateToCheckpointingProps,
+  mapDispatchToCheckpointingProps
+)(Checkpointing);
+
+const mapStateToSelectBatchIntervalRangeProps = (state) => {
+  return {
+    value: state.batchInterval.slice(0, -1),
+    options: BATCH_INTERVAL_RANGE
+  };
+};
+const mapDispatchToSelectBatchIntervalRangeProps = (dispatch) => {
+  return {
+    onChange: (e) => {
+      dispatch({
+        type: PipelineConfigurationsActions.SET_BATCH_INTERVAL_RANGE,
+        payload: { batchIntervalRange: e.target.value }
+      });
+    }
+  };
+};
+
+const SelectBatchIntervalRange = connect(
+  mapStateToSelectBatchIntervalRangeProps,
+  mapDispatchToSelectBatchIntervalRangeProps
+)(SelectWithOptions);
+
+const mapStateToSelectBatchIntervalUnitProps = (state) => {
+  return {
+    value: state.batchInterval.slice(-1),
+    options: BATCH_INTERVAL_UNITS
+  };
+};
+const mapDispatchToSelectBatchIntervalUnitProps = (dispatch) => {
+  return {
+    onChange: (e) => {
+      dispatch({
+        type: PipelineConfigurationsActions.SET_BATCH_INTERVAL_UNIT,
+        payload: { batchIntervalUnit: e.target.value }
+      });
+    }
+  };
+};
+
+const SelectBatchIntervalUnit = connect(
+  mapStateToSelectBatchIntervalUnitProps,
+  mapDispatchToSelectBatchIntervalUnitProps
+)(SelectWithOptions);
+
+const BatchInterval = () => {
```

Comment:
Not a huge deal, but it would be better to separate these components into their own files